core/src/test/scala/org/apache/spark/util/SizeEstimatorSuite.scala

```diff
@@ -214,6 +216,10 @@ class SizeEstimatorSuite
   }
 
   test("class field blocks rounding on 64-bit VM without useCompressedOops") {
+    System.setProperty(TEST_USE_COMPRESSED_OOPS_KEY, "false")
```

Comment:
Since this PR is all about explicit re-initialization, should we add `System.setProperty("os.arch", "amd64")` to this block as well?
Actually would it be a good idea to wrap this sequence into a helper method or something?

Reply:
I think that `System.setProperty("os.arch", "amd64")` has already done at `beforeEach()`.

Reply:
In which case it'd be better to keep all the relevant test cases consistent by removing those explicit initialization lines in a couple of other test cases in the same file?

Reply:
I think we should probably be as explicit as possible about what each test is doing. Let me add a follow up commit using a helper function and you can see what you think.